### PR TITLE
[rapids] bringing action up to date

### PR DESCRIPTION
**Pull request by cjac**
* adding functions to check running environment OS
* set rapids version to 23.12
* set cuda version to 11.8
* remove cudf variables - they are unused
* set xgboost version to 2.0.3
* created variable to store scala version for future move from 2.12 to 2.13
* setting python_ver by operating system

### CODE REVIEW

**Comment by cjac** [comment]
Manually tested on
❌
* 2.0-ubuntu18 — ✅
* 2.1-ubuntu20 — ✅
* 2.2-ubuntu22 — ✅
* 2.2-rocky9 — ✅
* 2.2-debian12 — ✅

**Comment by cjac** [comment]
this may depend on #1200 since I developed those changes as I was exercising these manual test runs.

**Comment by prince-cs** [comment]
`Step #5 - "dataproc-2.1-debian11-tests": 2024-07-22T14:53:01.119357965Z NameError: name 'metadata' is not defined`

https://github.com/GoogleCloudDataproc/initialization-actions/blob/master/rapids/test_rapids.py#L98

The presubmits seems to be failing because of this.

File: rapids/rapids.sh [hunk 1]
```diff
@@ -2,27 +2,79 @@
...
 
 readonly SPARK_VERSION_ENV=$(spark-submit --version 2>&1 | sed -n 's/.*version[[:blank:]]\+\([0-9]\+\.[0-9]\).*/\1/p' | head -n1)
 readonly DEFAULT_SPARK_RAPIDS_VERSION="22.10.0"
 
 if [[ "${SPARK_VERSION_ENV%%.*}" == "3" ]]; then
-  readonly DEFAULT_CUDA_VERSION="11.5"
-  readonly DEFAULT_CUDF_VERSION="22.10.0"
-  readonly DEFAULT_XGBOOST_VERSION="1.6.2"
-  readonly DEFAULT_XGBOOST_GPU_SUB_VERSION="0.3.0"
+  readonly DEFAULT_CUDA_VERSION="11.8"
```
Comment (prince-cs) [inline rapids/rapids.sh:73]: In GPU Driver we set the default cuda version to be 12.4. But, here it is 11.8. Any specific reason?

**Review by prince-cs — CHANGES_REQUESTED** [review]

File: rapids/test_rapids.py
Comment (prince-cs) [inline rapids/test_rapids.py:1]: Line no. 99: metadata is not required and is also not defined. Therefore, breaking the test flow.

**Review by prince-cs — CHANGES_REQUESTED** [review]

File: rapids/rapids.sh (hunk 1 above)
Comment (cjac) [reply rapids/rapids.sh:73]: 11.8 is the maximum supported by mamba package, 'cudatools', which is ultimately the goal of this script.

File: rapids/test_rapids.py
Comment (cjac) [reply rapids/test_rapids.py:1]: It looks like it is required and must have been left out in a previous commit.  fixed maybe.  I'll gcbrun to see.

File: rapids/test_rapids.py [hunk 2]
```diff
@@ -52,13 +53,13 @@ def verify_spark_job(self):
...
-                            ("STANDARD", ["m"], GPU_P100, "standalone"))
+  @parameterized.parameters(("STANDARD", ["m", "w-0"], GPU_T4, None),
+                            ("STANDARD", ["m", "w-0"], GPU_T4, "yarn"),
+                            ("STANDARD", ["m"], GPU_T4, "standalone"))
   def test_rapids_dask(self, configuration, machine_suffixes, accelerator,
                        dask_runtime):
-    if self.getImageOs() == 'rocky':
-      self.skipTest("Not supported in Rocky Linux-based images")
+    # if self.getImageOs() == 'rocky':
+    #   self.skipTest("Not supported in Rocky Linux-based images")
```
Comment (cyxxy) [inline rapids/test_rapids.py:62]: please just delete these.

File: rapids/test_rapids.py [hunk 3]
```diff
@@ -81,16 +82,18 @@ def test_rapids_dask(self, configuration, machine_suffixes, accelerator,
...
                                                machine_suffix))
 
-  @parameterized.parameters(("SINGLE", ["m"], GPU_P100),
-                            ("STANDARD", ["w-0"], GPU_P100))
+  @parameterized.parameters(("SINGLE", ["m"], GPU_T4),
+                            ("STANDARD", ["w-0"], GPU_T4))
   def test_rapids_spark(self, configuration, machine_suffixes, accelerator):
-    if self.getImageOs() == 'rocky':
-      self.skipTest("Not supported in Rocky Linux-based images")
+    # if self.getImageOs() == 'rocky':
```
Comment (cyxxy) [inline rapids/test_rapids.py:88]: ditto

File: rapids/test_rapids.py [hunk 4]
```diff
@@ -108,12 +111,12 @@ def test_rapids_spark(self, configuration, machine_suffixes, accelerator):
...
     self.verify_spark_job()
 
-  @parameterized.parameters(("STANDARD", ["w-0"], GPU_P100, "11.2"))
+  @parameterized.parameters(("STANDARD", ["w-0"], GPU_T4, "12.4"))
   def test_non_default_cuda_versions(self, configuration, machine_suffixes,
                                      accelerator, cuda_version):
 
-    if self.getImageOs() == 'rocky':
-      self.skipTest("Not supported in Rocky Linux-based images")
+#    if self.getImageOs() == 'rocky':
```
Comment (cyxxy) [inline rapids/test_rapids.py:118]: ditto

**Review by medb — APPROVED** [review]

**Review by cyxxy — APPROVED** [review]

**Pull request merged**